ibis/common/grounds.py

```diff
@@ -47,7 +48,14 @@ def __new__(metacls, clsname, bases, dct, **kwargs):
         # collection type annotations and convert them to validators
         module_name = dct.get('__module__')
         annotations = dct.get('__annotations__', {})
-        typehints = evaluate_annotations(annotations, module_name)
+
+        # pull in locals from the caller, purely for type annotation evaluation
+        #
+        # not great, but likely can't be avoided until
+        # https://peps.python.org/pep-0649/ is available
+        localns = sys._getframe(1).f_locals
```

Comment:
This wouldn't work for more nested function definitions.

I think we can restrict annotable to work only with globally defined classes for now. There are two test cases relying on local namespace in `ibis/common/tests/test_egraph.py` where we can pull out the classes to be module level definitions.

Reply:
Removed, and updated the tests.